homeassistant/components/dormakaba_dkey/config_flow.py

```diff
@@ -43,48 +43,14 @@ def __init__(self) -> None:
     async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
-        """Handle the user step to pick discovered device."""
-        errors: dict[str, str] = {}
-
-        if user_input is not None:
-            address = user_input[CONF_ADDRESS]
-            await self.async_set_unique_id(address, raise_on_progress=False)
-            # Guard against the user selecting a device which has been configured by
-            # another flow.
-            self._abort_if_unique_id_configured()
-            self._discovery_info = self._discovered_devices[address]
-            return await self.async_step_associate()
-
-        current_addresses = self._async_current_ids()
-        for discovery in async_discovered_service_info(self.hass):
-            if (
-                discovery.address in current_addresses
-                or discovery.address in self._discovered_devices
-                or not device_filter(discovery.advertisement)
-            ):
-                continue
-            self._discovered_devices[discovery.address] = discovery
-
-        if not self._discovered_devices:
-            return self.async_abort(reason="no_devices_found")
-
-        data_schema = vol.Schema(
-            {
-                vol.Required(CONF_ADDRESS): vol.In(
-                    {
-                        service_info.address: (
-                            f"{service_info.name} ({service_info.address})"
-                        )
-                        for service_info in self._discovered_devices.values()
-                    }
-                ),
-            }
-        )
-        return self.async_show_form(
-            step_id="user",
-            data_schema=data_schema,
-            errors=errors,
-        )
+        """Handle the user step.
+
+        The user will be shown a list of matching discovery flows automatically,
+        so we just abort a user flow.
+        """
+        if self._async_in_progress(include_uninitialized=True):
+            return self.async_abort(reason="no_additional_devices_found")
+        return self.async_abort(reason="no_devices_found")
```

Comment:
What if the discovery hasn't run yet? Then we should be able to force a discovery here, I think.

I don't think it's helpful for the user to just abort their effort to add the integration.

Reply:
I think that should be handled by improving the "add integration" dialog for bluetooth integrations which rely on Home Assistant's bluetooth discovery filter.

It could for example be updated so it shows a spinner + a text saying "Looking for compatible bluetooth devices", with the list of matching discovery flows updating while the dialog is open.

Reply:
I re-added the user flow for now so this integration is similar to other bluetooth integrations.